### ISSUE TITLE
Adventure Mode Showing what you DON'T have in your world. 

### ISSUE THREAD

**Issue by ethaldeman** [hzla/Remnant-World-Analyzer#5]
It seems after you updated the tool for the Corsus DLC, the Tool will now display what areas/bosses you DON'T have in that adventure. I don't know 100% how the items are working in the Adventure, but its seems it does show items you don't have in that Adventure. I believe the Campaign is still showing what area/bosses/items you do have.

**Comment by Troxbanv** [comment]
Looking at the file and making a few changes to the logic I am able to get the adventure world to work.  I built it in Java as that is what I am familiar with.  I looked at the lines one by one and looked at the location of the first instance of "Quest_AdventureMode".  That line will have all of the adventure mode information in it, I have run it around 10 times so far and it is accurate.  I would be happy to post the logic if you would like to look and hopefully it will help you out, it is based quite a bit on the logic you have so I think it may be easy to implement, I am just not a JavaScript guy.  Love the tool you have built and keep up the great work.

**Comment by Sale3054** [comment]
> Looking at the file and making a few changes to the logic I am able to get the adventure world to work. I built it in Java as that is what I am familiar with. I looked at the lines one by one and looked at the location of the first instance of "Quest_AdventureMode". That line will have all of the adventure mode information in it, I have run it around 10 times so far and it is accurate. I would be happy to post the logic if you would like to look and hopefully it will help you out, it is based quite a bit on the logic you have so I think it may be easy to implement, I am just not a JavaScript guy. Love the tool you have built and keep up the great work.

Can you post what you made or reference the err in logic? I will fix.

**Comment by Mini-Placzek** [comment]
> Looking at the file and making a few changes to the logic I am able to get the adventure world to work. I built it in Java as that is what I am familiar with. I looked at the lines one by one and looked at the location of the first instance of "Quest_AdventureMode". That line will have all of the adventure mode information in it, I have run it around 10 times so far and it is accurate. I would be happy to post the logic if you would like to look and hopefully it will help you out, it is based quite a bit on the logic you have so I think it may be easy to implement, I am just not a JavaScript guy. Love the tool you have built and keep up the great work.

Would be interested too. Also used this project as base but never fully understood how to parse the save file properly. Could help to fix the javascript which broke after the Corsus update.

**Comment by Troxbanv** [comment]
Sure I posted my code to [https://github.com/Troxbanv/Remnant_File_Analyzer](url).  The biggest change that I made was to the logic in finding the adventure and campaign worlds.  In my code if you look at the reader class under the readFile method I loop through the file instead of looking at the entire thing at once and if the line contains certain keywords I mark the location of the instance and the one with the lowest value is the line that contains all of the information for either the campaign or the adventure world.  I have been using this with a couple of friends and for all of us this has worked every time.  Thanks for converting the code to javascript, again I am more of a Java and C guy, I can read javascript but writing it I make way too many errors.  If you need anything else please feel free to ask and I will do my best to assist.

**Comment by Mini-Placzek** [comment]
> Sure I posted my code to [https://github.com/Troxbanv/Remnant_File_Analyzer](url). The biggest change that I made was to the logic in finding the adventure and campaign worlds. In my code if you look at the reader class under the readFile method I loop through the file one line at a time instead of looking at the entire thing at once and if the line contains certain keywords I mark the location of the instance and the one with the lowest value is the line that contains all of the information for either the campaign or the adventure world. I have been using this with a couple of friends and for all of us this has worked every time. Thanks for converting the code to javascript, again I am more of a Java and C guy, I can read javascript but writing it I make way too many errors. If you need anything else please feel free to ask and I will do my best to assist.

Thanks for sharing and great job man <3. Will start working on this now. Looks pretty straight forward so shouldn't too long. Will come back to you if I have any questions.

**Comment by Sale3054** [comment]
```function getAdventureEntries(arr)
{
  arr = arr.split("\n")
  tempList = []
  for(i = 0; i < arr.length; i++)
  {
    arrWord = String(arr[i])
    if (arrWord.includes('Adventure') === true)
    {
      if(tempList.includes(arrWord) === false)
      {
        tempList.push(arr[i])

      }
    }
  }
  return tempList[1]
}
`

**Comment by Sale3054** [comment]
![before-after](https://user-images.githubusercontent.com/22930880/83055059-92246100-a010-11ea-8ec2-cdc1e168e73a.png)

**Comment by Sale3054** [comment]
I will note that what's throwing off the author's solution is that there is a chunk of "adventure text" at the top of each save which, seemingly, contains all options that aren't present in your current save.

**Comment by Sale3054** [comment]
Looks like I can't create a pull request, so you can find my fix at my repo: 
https://github.com/Sale3054/RemnantWorldAnalyzerFix

**Comment by Troxbanv** [comment]
> I will note that what's throwing off the author's solution is that there is a chunk of "adventure text" at the top of each save which, seemingly, contains all options that aren't present in your current save.

Yes, that is why I changed the logic to find the smallest index of the phrase Quest_AdventureMode, even though there are many occurrences of that phrase the one with the smallest value has as of the Corsus update been the line which contains all of the adventure world items.

**Comment by Troxbanv** [comment]
> > Sure I posted my code to [https://github.com/Troxbanv/Remnant_File_Analyzer](url). The biggest change that I made was to the logic in finding the adventure and campaign worlds. In my code if you look at the reader class under the readFile method I loop through the file one line at a time instead of looking at the entire thing at once and if the line contains certain keywords I mark the location of the instance and the one with the lowest value is the line that contains all of the information for either the campaign or the adventure world. I have been using this with a couple of friends and for all of us this has worked every time. Thanks for converting the code to javascript, again I am more of a Java and C guy, I can read javascript but writing it I make way too many errors. If you need anything else please feel free to ask and I will do my best to assist.
> 
> Thanks for sharing and great job man <3. Will start working on this now. Looks pretty straight forward so shouldn't take too long. Will come back to you if I have any questions.

Great, hope it is an easy fix and look forward to seeing what the fix is.  Have been interested in Javascript just never have time to really dig into it.
